### ISSUE TITLE
Legion Toolkit starts running on the dGPU when I switch to Balanced or Performance mode in Hybrid mode

### ISSUE THREAD

**Issue by Undervoltologist** [BartoszCichecki/LenovoLegionToolkit#74]
Basically just the title, please fix it, because the Deactivate button won't work when it starts running on the dGPU, and it's too bad for the battery life.

**Comment by BartoszCichecki** [comment]
You, will need to provide more detail.

Toolkit does not activate the dGPU, because there is no reason to do it, regardless if you are in Quite, Balance or Performance mode.

Deactivate button will be active when three conditions are met:
1. There there are at least two GPUs in the system - Hybrid mode is on,
2. There are no displays connected to dGPU - so basically in case of Legion 5 Pro, HDMI and DisplayPort is not used,
3. At least one process is running on dGPU - this information is returned directly from NVAPI.

Basically it means that if you have no external displays and you have Hybrid mode turned on, then if the Deactivate button is disabled it means that nVidia drivers report there are no processes running on the dGPU.

How did you verify that the discrete GPU is running?

**Comment by Undervoltologist** [comment]
The Nvidia tray icon, Legion Toolkit shows up there, and the EC register 0xBD, it changes to 80 when the Nvidia GPU is running. 
Deactivate button is not disabled, it shows the Nvidia GPU is active and when I click on Deactivate, Toolkit can't remove itself, keeps running. https://photos.app.goo.gl/eakVMsqoa1RtuDiq7

**Comment by BartoszCichecki** [comment]
Interesting, I am not sure why Windows or the nVidia driver runs Toolkit on dGPU on your machine. I can't repro this.

Did you customize any settings in the driver? Can you try forcing Toolkit to run od iGPU in Windows settings?

**Comment by Undervoltologist** [comment]
No customizations related with the Toolkit, also at this point, it just randomly turns on the dGPU, basically it will launch in iGPU, but after some time, it will switch over to the dGPU. :/ I will try forcing it though.

**Comment by Undervoltologist** [comment]
Yup thanks, seems fixed after forcing it to run with dGPU!
